### PR TITLE
Getting fprime-vxworks to compile for Vx7

### PR DESCRIPTION
**Pull request by kevin-f-ortega**
This PR is to get fprime-vxworks compiling for both Vx6 and Vx7.

### CODE REVIEW

**Comment by kevin-f-ortega** [comment]
Depends on https://github.com/nasa/fprime/pull/1649

File: cmake/platform/types/PlatformTypes.hpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/**
```
Comment (LeStarch) [inline cmake/platform/types/PlatformTypes.hpp]: Is this exactly like the linux one?

**Review by LeStarch — COMMENTED** [review]
Looks right to me.  Does it work?

**Comment by kevin-f-ortega** [comment]
I was able to compile an empty deployment (that means no topology or instance, just a main.cpp) for Vx7 using the PPC toolchain.

File: fprime-vxworks/VxWorks/VxWorks7/WatchdogTimer.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    static int myWatchdogCallback(
+        int arg //!< arg is the WatchdogTimer instance that initiated the wdStart
+    )
+    {
+        WatchdogTimer *comp = reinterpret_cast<WatchdogTimer*>(arg);
+        FW_ASSERT(comp != nullptr);
+        WatchdogTimer::WatchdogCb cb = comp->getCallback();
+        cb(comp->getParameter());
+        return OK;
+    }
```
Comment (kevin-f-ortega) [inline fprime-vxworks/VxWorks/VxWorks7/WatchdogTimer.cpp]: @LeStarch , take a look at this piece of code. I added this wrapper to get around the different FUNCPTR definitions between Vx6 and Vx7. This wrapper needs access to some private member variables, hence the https://github.com/nasa/fprime/pull/1649 PR.

File: fprime-vxworks/VxWorks/VxWorks7/Task.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    //!< with the provides arg. This wrapper function is expected to be
+    //!< passed in to taskCreate.
+    static int myRoutineWrapper(
+        Task::taskRoutine routine, //!< Routine that should run in a thread
+        void* arg //!< Argument passed to the routine
+    )
+    {
+        routine(arg);
+        return OK;
+    }
```
Comment (kevin-f-ortega) [inline fprime-vxworks/VxWorks/VxWorks7/Task.cpp]: @LeStarch , I use the same function-wrapper strategy here.

File: fprime-vxworks/VxWorks/VxWorks7/WatchdogTimer.cpp (hunk 2 above)
Comment (LeStarch) [reply fprime-vxworks/VxWorks/VxWorks7/WatchdogTimer.cpp]: I see. Since callback and parameter aren't natively passable to the Os watchdog code, then you have to wrap it.  However,  it seems to me:

```
void expire() {
    this->callback(this->parameter);
}
```

Should be wrapped in a function such that all your above code has to do is call expire.  That way the implementation doesn't require exposing private members like getter/setter.

File: cmake/platform/VxWorks.cmake [hunk 4]
```diff
@@ -6,4 +6,4 @@ add_definitions(-D_WRS_KERNEL)
 set(FPRIME_USE_BAREMETAL_SCHEDULE OFF)
 
 # Add VxWorks specific headers into the system
-include_directories(SYSTEM "${FPRIME_FRAMEWORK_PATH}/Fw/Types/VxWorks")
+include_directories(SYSTEM "${CMAKE_CURRENT_LIST_DIR}/types")
```
Comment (kevin-f-ortega) [inline cmake/platform/VxWorks.cmake]: Add directory for Vx6 and Vx7. Move Vx7 types to Vx7 and copy Vx7 types to Vx6. This "Vx6" types file will be edited in a separate PR.

**Comment by kevin-f-ortega** [comment]
This PR is out of date. `devel` has already been updated to compile for vx7.
